src/actions/requests.rs

```diff
@@ -82,15 +78,13 @@ impl<'a> Action<'a> for Symbols {
 impl<'a> RequestAction<'a> for Symbols {
     type Response = Vec<SymbolInformation>;
     fn handle<O: Output>(&mut self, _id: usize, params: Self::Params, ctx: &mut ActionContext, _out: O) -> Result<Self::Response, ()> {
-        let t = thread::current();
         let ctx = ctx.inited();
         let file_path = parse_file_path!(&params.text_document.uri, "symbols")?;
 
         let analysis = ctx.analysis.clone();
 
-        let rustw_handle = thread::spawn(move || {
+        let receiver = receive_from_thread(move|| {
```

Comment:
style nit: I'd keep `move ||` space-separated version to be more consistent with the style used in other places in the codebase

Reply:
Will do